hbase-server/src/main/java/org/apache/hadoop/hbase/master/assignment/AssignmentManager.java

```diff
@@ -616,6 +616,24 @@ public long assign(RegionInfo regionInfo) throws IOException {
     return assign(regionInfo, null);
   }
 
+  public Future<byte[]> assignAsync(RegionInfo regionInfo, ServerName sn) throws IOException {
```

Comment:
Most of this method's code is a duplicate  of the assign() above, the common part can be wrapped in a private method.

Reply:
If we are adding a new public method, can we have doc for these two new methods?

Reply:
You are absolutely right. I'll fix this.